### DISCUSSION TITLE
Compact storage of conformers

### DISCUSSION THREAD

**Discussion by tdudgeon** [rdkit/rdkit#5551] [Q&A]
Has anyone looked into good ways to store conformers of molecules in very compact manner (minimising storage needs).
Using SDF is clearly very inefficient.
Does not need to be portable across other toolkits.

**Comment by ptosco** [comment]
If you do not need portability, you could simply pickle the RDKit `Chem.Mol` object. CXSMILES also offers a compact way to store coordinates.

**Comment by tdudgeon** [comment]
Thanks Paolo. Presumably using a pickle is potentially not compatible across different RDKit versions?
I had a try using cxsmiles and the files are just over 3x smaller, so that's useful but not spectacular.

**Reply by greglandrum** [reply]
@tdudgeon the pickle format is definitely backwards compatible. The current version of the RDKit will read pickles from every RDKit version that's ever been released.

**Comment by kienerj** [comment]
Usually depends what the prime driver is.  Storage or processing speed. In case of storage the obvious answer is compression and in general if speed doesn't matter lzma is generally one of the best for general usage. You can compress the sdf itself or the pickled molecule. In the later case probably worth a try to test compression factor. 

See also python-blosc for compression of pickled data. pickle + blsosc just from reading seems to likley offer the best combination of storage and speed but again you will need to play around and find a suitable combination.

**Source:**
https://stackoverflow.com/questions/57983431/whats-the-most-space-efficient-way-to-compress-serialized-python-data
which contains a link to
https://github.com/Blosc/python-blosc
